naga/src/compact/mod.rs

```diff
@@ -129,62 +203,23 @@ pub fn compact(module: &mut crate::Module) {
         module_tracer.global_expressions_used.insert(constant.init);
     }
 
-    // We treat all named overrides as used by definition.
-    log::trace!("tracing named overrides");
-    for (handle, r#override) in module.overrides.iter() {
-        if r#override.name.is_some() {
-            log::trace!("tracing override {:?}", r#override.name.as_ref().unwrap());
-            module_tracer.overrides_used.insert(handle);
-            module_tracer.types_used.insert(r#override.ty);
-            if let Some(init) = r#override.init {
-                module_tracer.global_expressions_used.insert(init);
+    if preserve_unused {
+        // Treat all named overrides as used.
+        for (handle, r#override) in module.overrides.iter() {
+            if r#override.name.is_some() && !module_tracer.overrides_used.contains(handle) {
+                module_tracer.overrides_used.insert(handle);
+                module_tracer.types_used.insert(r#override.ty);
```

Comment:
suggestion(non-blocking): You can combine `contains` and `insert` if you use the `bool` return value of the latter:

```suggestion
        for (handle, r#override) in module.overrides.iter() {
            if r#override.name.is_some() && module_tracer.overrides_used.insert(handle) {
                module_tracer.types_used.insert(r#override.ty);
```

Reply:
Pushed a `fixup!` with 1ac00d07. LMKWYT!